client/app/bundles/course/level/components/LevelsTable.tsx

```diff
@@ -0,0 +1,158 @@
+import { FC } from 'react';
+import { defineMessages, FormattedMessage } from 'react-intl';
+import { Add, Delete, Undo } from '@mui/icons-material';
+import {
+  Button,
+  IconButton,
+  Table,
+  TableBody,
+  TableCell,
+  TableFooter,
+  TableHead,
+  TableRow,
+  TextField,
+  Tooltip,
+  Typography,
+} from '@mui/material';
+
+import useTranslation from 'lib/hooks/useTranslation';
+
+const translations = defineMessages({
+  levelHeader: {
+    id: 'course.level.Level.levelHeader',
+    defaultMessage: 'Level',
+  },
+  thresholdHeader: {
+    id: 'course.level.Level.thresholdHeader',
+    defaultMessage: 'Threshold',
+  },
+  addNewLevel: {
+    id: 'course.level.Level.addNewLevel',
+    defaultMessage: 'Add New Level',
+  },
+  orderedIncorrectly: {
+    id: 'course.level.Level.orderedIncorrectly',
+    defaultMessage:
+      'Levels will be sorted automatically when saved regardless of their order here.',
+  },
+});
+
+interface LevelsTableProps {
+  levels: { level: number; deleted: boolean }[];
+  canManage: boolean;
+  isSaving: boolean;
+  onAddLevel: () => void;
+  onThresholdChange: (index: number, newThreshold: string) => void;
+  onDeleteLevel: (index: number) => void;
+  onUndoDelete: (index: number) => void;
+}
+
+const getClassName = (deleted: boolean, index: number): string => {
+  if (deleted) return 'bg-red-100'; // Red tint when marked for deletion
+  return index % 2 === 0 ? 'bg-white' : 'bg-gray-100';
+};
+
+const LevelsTable: FC<LevelsTableProps> = ({
+  levels,
+  canManage,
+  isSaving,
+  onAddLevel,
+  onThresholdChange,
+  onDeleteLevel,
+  onUndoDelete,
+}) => {
+  const { t } = useTranslation();
+  const isThresholdOrderedCorrectly = (index: number): boolean =>
+    index === 0 || levels[index].level > levels[index - 1].level;
+
+  return (
+    <Table>
+      <TableHead>
+        <TableRow>
+          <TableCell>
+            <FormattedMessage {...translations.levelHeader} />
+          </TableCell>
+          <TableCell>
+            <FormattedMessage {...translations.thresholdHeader} />
+          </TableCell>
+          <TableCell />
+        </TableRow>
+      </TableHead>
+
+      <TableBody>
+        {levels.slice(1).map((row, index) => (
+          <TableRow
+            // Index can be used as key as levels are not reordered
+            // eslint-disable-next-line react/no-array-index-key
+            key={`level-row-${index}`}
+            className={getClassName(row.deleted, index)}
+          >
+            <TableCell>
+              <Typography>{index + 1}</Typography>
```

Comment:
Please make this `text-center`, align the title `Level` and also all the entries in that column to be centred

Reply:
I think this looks very weird, and its much better off being flush to the left

Reply:
I see. In that case, please align the title `Level` with its entry (I noticed that `Level` is a bit indented to the right compared to its entries below)

Reply:
Done, currently the tailwind styling used is `p-6` instead of using rem, because rem does not work for this MUI component